### PR TITLE
Use property

### LINKED ISSUE #119: Use Properties in Molecule
`molecular_info.py` defines many get and set methods. These should be `@property`.

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
This PR closes #119 by using the property decorator. `property` can [be](https://www.programiz.com/python-programming/property) [used](https://www.python-course.eu/python3_properties.php) to implement a "getter" and a "setter", a way for a user to get and set a variable from a class without needing to worry if the name of the actual variable changes. All that needs to stay the same is the name of the property.

Making this work required fixing a few test cases. I also fixed the OpenFermion-related test cases, which I had been ignoring for some time.

## User Notes
- [x] System and Molecule now use the property interface

## Checklist
- [x] Tests pass
- [x] Ready to go!

### CODE REVIEW

**Review by imagoulas — APPROVED** [review]
Approved. I just  have a couple of questions (see the relevant comments).

File: tests/fast_qk_test.py [hunk 1]
```diff
@@ -409,9 +409,9 @@ def test_H4_fast_qkd(self):
         Egs1 = alg1.get_gs_energy()
         self.assertLess(abs(Egs1-E_fci), 1.0e-4)
 
-        # Non-Trotteized (exact) SRQK
+        # Non-Trotterized (exact) SRQK
         alg0 = NTSRQK(mol, reference=ref)
-        alg0.run(s=6, dt = 0.8)
+        alg0.run(s=7, dt = 0.8)
```
[the remark this answers is not in the document]
Comment (JonathonMisiewicz) [reply tests/fast_qk_test.py:414]: We weren't converged to within the test tolerance. Increasing s is the equivalent of increasing the number of iterations.

File: src/qforte/adapters/molecule_adapters.py [hunk 2]
```diff
@@ -192,31 +192,29 @@ def create_openfermion_mol(**kwargs):
         qubit_hamiltonian = jordan_wigner(fermion_hamiltonian)
         qforte_hamiltonian = build_from_openfermion(qubit_hamiltonian)
 
-    qforte_mol.set_hf_reference(hf_reference)
-    qforte_mol.set_hamiltonian(qforte_hamiltonian)
-    qforte_mol.set_sq_hamiltonian( build_sqop_from_openfermion(fermion_hamiltonian) )
-    qforte_mol.set_sq_of_ham(fermion_hamiltonian)
```
[the remark this answers is not in the document]
Comment (JonathonMisiewicz) [reply src/qforte/adapters/molecule_adapters.py:198]: Correct.

**Comment by JonathonMisiewicz** [comment]
Force pushed to resolve merge conflict. I'm going ahead to merge.

**Pull request merged**